### DISCUSSION TITLE
HTTPS credential connection with N8N

### DISCUSSION THREAD

**Discussion by 2tmmkt** [devlikeapro/waha#680] [Q&A]
I had to scale my operation and migrated my system to WAHA PLUS and a NOWEB structure. Making this, I also tried to migratte my IP to HTTPS, but when I tried to create a credential in N8N it doesn't work.

I am using a server in VULTR and using docker-compose to code my docker updates. I used it to install HTTPS and it worked, but I had this problem with N8N credentials. All containers are in the same network. I also tried to make some nginx changes in the SSL certificate but no success.

Someone who uses N8N with WAHA can help me with this HTTPS problem? I didn't had any issues with HTTP, just with HTTPS. The error was in N8N WAHA credential was: DEPTH_ZERO_SELF_SIGNED_CERT.

Thank you in advance!

**Comment by devlikepro** [comment]
Hi! Try installing using nginx reverse proxy with certbot
Here's the steps https://waha.devlike.pro/docs/how-to/install/#additional-steps

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by 2tmmkt** [reply]
Thank you for your support. I tried this, but certbot can't access the challenge. I am using a subdomain of a domain that is connected to cloudron, I suspect that it maybe giving some conflict and is not allowing certbot to access the subdomain.

I bought a new domain just for this and the error still the same.

The error I get by trying to install the certificate with certbot is:

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
certbot is already the newest version (2.9.0-1).
0 upgraded, 0 newly installed, 0 to remove and 8 not upgraded.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
python3-certbot-nginx is already the newest version (2.9.0-1).
0 upgraded, 0 newly installed, 0 to remove and 8 not upgraded.
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Requesting a certificate for wahaengrena.com.br

Certbot failed to authenticate some domains (authenticator: webroot). The Certificate Authority reported these problems:
  Domain: wahaengrena.com.br
  Type:   unauthorized
  Detail: 2001:19f0:ac00:4ef5:5400:5ff:fe1a:f484: Invalid response from http://wahaengrena.com.br/.well-known/acme-challenge/geH2G8E0w0z1FSc-27b9-zB7yy70mT_NoxG6-4pxe78: 404

Hint: The Certificate Authority failed to download the temporary challenge files created by Certbot. Ensure that the listed domains serve their content from the provided --webroot-path/-w and that files created there can be downloaded from the internet.

Some challenges have failed.
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /var/log/letsencrypt/letsencrypt.log or re-run Certbot with -v for more details.

How do I get this solved? I tried somethings like creating a test file in the challenge folder and tried to reach it through the domain but it didn't work

ps: the domain is pointing to my vultr server.

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
looks like the HTTP server is occupied by https://www.cloudron.io/
please check their docs to see how to handle your case

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
or consider using clean, dedicated for WAHA Ubuntu based server, never had a problem with this setup :) 

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
